### PR TITLE
feat: [CCM-7436]: Perspective Preferences

### PR DESCRIPTION
**Pull request by rajarshee-harness**
##### Summary:

Allow users to set perspective preferences in create perspective flow and perspective time series chart.

##### Jira Links:

https://harness.atlassian.net/browse/CCM-7436

##### Screenshots:

Screenshot 2022-06-22 at 11 59 24 AM

Screenshot 2022-06-22 at 12 00 05 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by rajarshee-harness** [comment]
trigger cypress

File: src/modules/75-ce/components/PerspectivePreferences/PerspectivePreferences.module.scss [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  display: flex;
+
+  .labelContainer {
+    display: flex;
+    align-items: center;
+    margin-bottom: var(--spacing-medium);
+
+    .prefLabel {
+      font-weight: 500;
+      font-size: 12px;
```
Comment (jenil-harness) [inline src/modules/75-ce/components/PerspectivePreferences/PerspectivePreferences.module.scss:19]: `Text` component here.

File: src/modules/75-ce/pages/perspective-details/PerspectiveDetailsPage.tsx [hunk 2]
```diff
@@ -413,6 +430,9 @@ const PerspectiveDetailsPage: React.FC = () => {
   const { licenseInformation } = useLicenseStore()
   const isFreeEdition = licenseInformation['CE']?.edition === ModuleLicenseType.FREE
 
+  const isClusterDatasource =
+    perspectiveData?.dataSources?.length === 1 && perspectiveData?.dataSources.includes('CLUSTER')
+
```
Comment (jenil-harness) [inline src/modules/75-ce/pages/perspective-details/PerspectiveDetailsPage.tsx:435]: knit: Move it to a util fn.

File: src/modules/75-ce/pages/perspective-details/PerspectiveDetailsPage.tsx [hunk 3]
```diff
@@ -522,3 +552,51 @@ const PerspectiveDetailsPage: React.FC = () => {
...
+const PreferencesDropDown: React.FC<{
+  preferences: QlceViewPreferencesInput
+  setPreferences: React.Dispatch<React.SetStateAction<QlceViewPreferencesInput>>
+  showIncludeUnallocatedCost: boolean
+}> = ({ preferences, setPreferences, showIncludeUnallocatedCost }) => {
+  const { getString } = useStrings()
+
+  return (
+    <Popover
+      interactionKind="click"
```
Comment (jenil-harness) [inline src/modules/75-ce/pages/perspective-details/PerspectiveDetailsPage.tsx:565]: import it from `blueprintjs`

**Review by jenil-harness — APPROVED** [review]

**Review by jyotiarora-harness — APPROVED** [review]

File: src/modules/75-ce/pages/perspective-details/PerspectiveDetailsPage.tsx (hunk 2 above)
Comment (rajarshee-harness) [reply src/modules/75-ce/pages/perspective-details/PerspectiveDetailsPage.tsx:435]: There already was an util for this (`clusterInfoUtil` in `perspectiveUtils`). Will use that.

File: src/modules/75-ce/components/PerspectivePreferences/PerspectivePreferences.module.scss (hunk 1 above)
Comment (rajarshee-harness) [reply src/modules/75-ce/components/PerspectivePreferences/PerspectivePreferences.module.scss:19]: This class is passed to `Switch` to override it's label style, so can't use `Text`.

**Pull request merged**